stellar/contract/round/src/internal.rs

```diff
@@ -0,0 +1,802 @@
+    },
+    funding_writer::{add_total_funding, read_total_funding},
+    methods::RoundTrait,
+    pair::{get_all_pairs, get_pair_by_index, get_random_pairs},
+    project_registry_writer::{read_project_contract, write_project_contract},
+    round_writer::{is_initialized, read_round_info, write_round_info},
+    storage::extend_instance,
+    token_writer::{read_token_address, write_token_address},
+    voter_writer::{
+        add_to_black_list, add_to_white_list, is_black_listed, is_white_listed,
+        remove_from_black_list, remove_from_white_list,
+    },
+    voting_writer::{
+        add_voting_result, find_voting_result, get_voting_state, increment_voting_count,
+        set_voting_state,
+    },
+};
+use loam_sdk::soroban_sdk::{self, contract, contractimpl, token::TokenClient, String, Vec};
+use loam_sdk::soroban_sdk::{Address, Env};
+loam_sdk::import_contract!(project_registry);
+
+#[contract]
+pub struct Round;
+
+#[contractimpl]
+impl RoundTrait for Round {
+    fn init(
+        env: &Env,
+        owner: Address,
+        token_address: Address,
+        registry_address: Address,
+        round_detail: CreateRoundParams,
+    ) {
+        owner.require_auth();
+
+        assert!(
+            round_detail.start_time < round_detail.end_time,
+            "Round start time must be less than round end time"
+        );
+        assert!(
+            round_detail.application_start_time < round_detail.application_end_time,
+            "Round application start time must be less than round application end time"
+        );
+        assert!(
+            round_detail.end_time > round_detail.application_start_time,
+            "Round end time must be greater than round application start time"
+        );
+        assert!(round_detail.amount > 0, "Amount must be greater than 0");
+        assert!(
+            !round_detail.admins.is_empty(),
+            "Round admins must not empty"
+        );
+        assert!(
+            round_detail.admins.len() < 5,
+            "Round admins must be less than 5"
+        );
+
+        let round_init = is_initialized(env);
+
+        assert!(!round_init, "Round already initialized");
+
+        let mut num_picks_per_voter = 2;
+
+        if round_detail.num_picks_per_voter.is_some() {
+            num_picks_per_voter = round_detail.num_picks_per_voter.unwrap();
+            assert!(
+                num_picks_per_voter > 0,
+                "Number of picks per voter must be greater than 0"
+            );
+            assert!(
+                num_picks_per_voter <= 10,
+                "Number of picks per voter must be less than or equal 10"
+            );
+        }
+
+        assert!(
+            round_detail.contact.len() <= 10,
+            "Contact must be less than 10"
+        );
+        assert!(
+            round_detail.image_url.len() <= 200,
+            "Image URL must be less than 200 characters. Use IPFS Hash Only"
+        );
```

Comment:
round does not have an image, no?

https://www.figma.com/design/m8CMab6IoqEXFiebfYQu2B/GrantPicks-(Budget-Boxing)?node-id=1379-8547&t=uMff5Mgcq5aOkpns-0